pontoon/projects/management/commands/send_approved_notifications.py

```diff
@@ -0,0 +1,88 @@
+from collections import defaultdict
+from datetime import timedelta
+from urllib.parse import urlencode
+
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+from django.urls import reverse
+from django.utils import timezone
+from notifications.signals import notify
+from pontoon.base.models import Translation
+
+
+class Command(BaseCommand):
+    help = "Notify translators about their newly approved/rejected suggestions"
+
+    def get_description(self, author, locale, project, approved, rejected):
+        url = reverse(
+            "pontoon.translate",
+            kwargs={
+                "locale": locale.code,
+                "project": project.slug,
+                "resource": "all-resources",
+            },
+        )
+        if len(approved) == 1 and len(rejected) == 0:
+            url += "?" + urlencode({"string": approved[0]})
+        elif len(approved) == 0 and len(rejected) == 1:
+            url += "?" + urlencode({"string": rejected[0]})
+        else:
+            url += "?" + urlencode({"author": author.email})
+
+        if len(approved) == 0:
+            if len(rejected) == 1:
+                msg = "one of your suggestions was rejected"
+            else:
+                msg = f"{len(rejected)} of your suggestions were rejected"
+        else:
+            if len(approved) == 1:
+                msg = "one of your suggestions was approved"
+            else:
+                msg = f"{len(approved)} of your suggestions were approved"
+            if len(rejected) == 1:
+                msg += ", and one was rejected"
+            elif len(rejected) > 1:
+                msg += f", and {len(rejected)} were rejected"
```

Comment:
I would simplify the logic and only list project-locales with reviewed strings, same as we do in the unreviewed suggestions notifications.

Count is useful, I agree, but I'd rather not mess with the sentences, and instead use a graphic, e.g. <img width="54" alt="Screenshot 2022-03-15 at 19 49 32" src="https://user-images.githubusercontent.com/626716/158452369-ead3cb33-7f1c-4112-9c7e-1082c2a89412.png"> 

Maybe do that in a follow-up, also for unreviewed suggestion notifications?

Reply:
The sentence construction isn't really complicated, it's just complex.

I think this and the discussion on due-date notifications is sounding like an impending review of all notifications.

Reply:
For unreviewed suggestions notifications we also send just one notification per user, here we send one per project-locale combination. (And add counts.)

I'm also curious what @flodolo thinks about the message structure.

Reply:
> I'm also curious what @flodolo thinks about the message structure.

It feels very complex to build. As the recipient of such message, it also forces my brain to do some extra work, because the structure changes depending on the data, and it's not as immediate to parse.

I'd be happy with something minimal (and less elegant/natural) like.

```
Some of your suggestions were reviewed:
* Approved: XX strings
* Rejected: XX strings
```

And drop approved/rejected if the number is 0.

Reply:
Thanks, Flod. What is your take on 1 notification per user vs one pre each project-locale-user combination?

Reply:
I think 1 notification would be ideal, as long as we can link to specific strings for each project in the message.

I also don't expect notifications to include a lot of projects, so maybe the end result is the same in most cases (for most users, it will be about 1 project).

Reply:
As it would appear from this discussion that the spec for the notification contents that I proposed in https://github.com/mozilla/pontoon/issues/2094#issuecomment-1066754638 isn't good enough after all, how should we proceed with re-specifying it?

Reply:
I suggest we follow the lead of Unreviewed suggestions notifications and send only 1 notification per user with the following content:

```
Your suggestions in the following projects have been reviewed:
* Firefox (X Approved, Y Rejected)
* Common Voice (Z Approved)
```

I would also implement a new `review_time` filter, similar to the `time` filter, and use it in the link.

In a followup, we could also add count to Unreviewed suggestions notifications.

Reply:
Okay; is this how the notification would look like for a single approved string in one project?

```
Your suggestion in the following project has been reviewed:
* Firefox (1 Approved)
```

That seems a bit clumsy. Would it make sense to use a different format if all the strings are in one project, like this?

```
Your suggestions in Firefox have been reviewed (2 Approved, 1 Rejected)
```

And with a special case for a single string:

```
Your suggestion in Firefox has been Approved
```

Also, if we're listing projects and not project+locale combos, how should we create the links if a user had strings reviewed for one project in more than one language?

Reply:
Personally, I prefer consistent message structure for readability (I can quickly scan the relevant info, which is project and number of strings review).

Review for multiple locales+projects within two cron job runs is an edge case, but we clearly need to deal with it somehow. Would it be easier to send separate notifications for each locale?

Reply:
Agreed with Flod about the message structure.

For Unreviewed, if more than one locales are affected, we list project for each locale and set the URL accordingly. We should show the locale identifier, as well. For example:

```
Your suggestions in the following projects have been reviewed:
* Firefox (Slovenian · sl): X Approved, Y Rejected
* Common Voice (french · fr): Z Approved
```

Reply:
Notification contents refactored as per above. Left out "in the following projects" as redundant, and leading to complexity in needing to match the plural case to the actual count of projects.